src/api/pool.go

```diff
@@ -0,0 +1,109 @@
+package api
+
+import (
+	"context"
+	"github.com/go-resty/resty/v2"
+	"golang.org/x/sync/errgroup"
+	"runtime"
+	"sync"
+)
+
+type Pool struct {
+	client          *Client         // http client
+	ctx             context.Context // context of the parallel work
+	workers         *errgroup.Group // error group -> if one worker fails, all will be stopped
+	counter         sync.WaitGroup  // detect when all requests are processed
+	sendersCount    int
+	processorsCount int
+	processor       func(pool *Pool, response *resty.Response) error
+	requests        chan *resty.Request
```

Comment:
`chan` -> kanal ... je to nieco ako `pipe`, z jednej strany sa zapisuje a z druhej cita, ... je to bezpecne na pouzite s `goroutines` -> na komunikaciu medzi roznymi vlaknami. Kazdy kanal ma svoj typ, ... v tomto pripade pointer na Request/Response.